### DISCUSSION TITLE
How to train using DistributedDataParallel with custom training script?

### DISCUSSION THREAD

**Discussion by haimat** [ultralytics/yolov5#6370] [General]
I call the YOLOv5 `train()` method from a custom Python application, i.e. not via `train.py` from command line as usual.  
What would be the best way to make use of the *DistributedDataParallel* mode in this case?

**Comment by glenn-jocher** [comment]
@haimat DDP is pretty complicated, which is why we provide a training script.

**Reply by haimat** [reply]
@glenn-jocher Thanks for the feedback, I will have a deeper look into it then!
